README.md

```diff
@@ -1 +1,68 @@
-# be-take-home-assignment-golang
+# Connected Freight BE Take-home Assignment (Golang)
+
+Hello there 👋! Thank you for taking this interview 🙏 and congratulations on making it this far 🎉.
+
+---
+
+There are 2 questions in total for this assignment, and should take no more than 90 mins to complete. They are organized in 2 folders:
+- [Question 1](https://github.com/connected-freight/be-take-home-assignment-golang/tree/main/question-1)
+- [Question 2](https://github.com/connected-freight/be-take-home-assignment-golang/tree/main/question-2)
+
+Each question folder has instructions and expectations. Please try to follow them and let us know, if you have any issues/concerns. You are only expected to spend 30 mins on Question 1, 60 mins on Question 2, and submit what you have done so far, to not take up too much of your time. You may take longer if you like 😊
+
+---
+
+Key things we are focusing on are:
+
+- [x] Follow best practices.
+- [x] Code readability.
+- [x] Code structure.
+- [x] Naming of parameters, functions, types, and fields.
+- [x] Appropriate use of comments.
+
+---
+
+## Getting started
+
+> IMPORTANT: Our automated scripts to prepare submissions for grading requires the original git history to be retained in your submitted repository. If the script is unable to automatically clone your repository, it will **not be graded and failed immediately**. Steps 1 to 3 will allow you to create a private repo with the original git history :)
+
+As GitHub does not allow the forking of public repos to private repos, what you need to do instead is to clone this repo into a new personal private repo.
+
+### 1. Create a new private repo
+When creating a new private repo in GitHub, name the repo any way you like (but preferably `cf-be-take-home-assignment-golang` if you want to follow the instructions below as it is), set it to private, and **do not initialize the repository with anything**.
+
+### 2. Clone this repo to your local filesystem
+```
+git clone https://github.com/connected-freight/be-take-home-assignment-golang.git cf-be-take-home-assignment-golang
+```
+
+### 3. Add your remote and push
+Assuming that you are using `cf-be-take-home-assignment-golang` as your repo name:
+```
+cd cf-be-take-home-assignment-golang
+git remote rename origin upstream
+git remote add origin https://github.com/<your-username>/cf-be-take-home-assignment-golang.git
+git push origin main
+```
+Now you can work with it just like any other GitHub repo.
+
+### 4. Setup development environment
+
+#### Install development tools
+The following tools are required:
+1. [Golang](https://go.dev/doc/install)
+2. A Go-compatible IDE (For example: [Visual Studio Code](https://code.visualstudio.com/))
+
+#### Run tests
+```
+go test -v ./...
+```
+
+---
+
+## Submission
+When you are ready to submit, push all of your changes, and add `nico@connectedfreight.com` as a collaborator to your repo for grading 😊
```

Comment:
@nico-cf are you going to to be the main grader for GoLang homework assignments? If so, should we change this?

Reply:
Sure, that's fine. I've changed the email address to my Github username.